src/components/DashboardHeader.tsx

```diff
@@ -7,13 +7,25 @@ import { getUserAgentApplication } from "../helpers/msal";
 
 
 export class DashboardHeader extends Component {
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            loggedUser: ""
+        };
+      }
+
     public onSignOut = async () => {
         const userAgentApplication = getUserAgentApplication();
         sessionStorage.clear();
         userAgentApplication.logout();
-        
       };
 
+    public componentDidMount = async () => {
```

Comment:
I'd suggest to switch to react hooks in functional components (useEffetc) rather than rely on these class methods:

https://medium.com/@felippenardi/how-to-do-componentdidmount-with-react-hooks-553ba39d1571#:~:text=The%20equivalent%20of%20componentDidMount%20in%20hooks%20is%20the%20useEffect%20function.&text=Functions%20passed%20to%20useEffect%20are,a%20second%20argument%20to%20it.